lib/jsdom/living/nodes/HTMLSlotElement-impl.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+const idlUtils = require("../generated/utils");
+const HTMLElement = require("../generated/HTMLElement");
+const HTMLElementImpl = require("./HTMLElement-impl").implementation;
+const { getRoot, assignSlotableForTree, findFlattenedSlotables } = require("../helpers/shadow-dom");
+
+class HTMLSlotElementImpl extends HTMLElementImpl {
+  constructor(args, privateData) {
+    super(args, privateData);
+    this._assignedNodes = [];
+  }
+
+  // https://html.spec.whatwg.org/#dom-slot-name
+  get name() {
+    const value = this.getAttribute("name");
+    return value || "";
+  }
+  set name(value) {
+    this.setAttribute("name", value);
+  }
+
+  _attrModified(name, value, oldValue) {
+    super._attrModified(name, value, oldValue);
+
+    // https://dom.spec.whatwg.org/#slot-name
+    if (name === "name" && value !== oldValue) {
+      if ((!value && oldValue === "") || (value === "" && !oldValue)) {
```

Comment:
Follow the spec more exactly here, with a series of `if` statements that do early returns. Collapsing them all into a big if statement makes it harder for me to understand the correspondence.